alacritty/src/cursor.rs

```diff
@@ -17,23 +17,29 @@
 use std::cmp;
 
 use alacritty_terminal::ansi::CursorStyle;
+use alacritty_terminal::config::CursorThickness;
 
 use font::{BitmapBuffer, Metrics, RasterizedGlyph};
 
-/// Width/Height of the cursor relative to the font width
-pub const CURSOR_WIDTH_PERCENTAGE: f64 = 0.15;
-
 pub fn get_cursor_glyph(
     cursor: CursorStyle,
     metrics: Metrics,
     offset_x: i8,
     offset_y: i8,
     is_wide: bool,
+    cursor_thickness: CursorThickness,
 ) -> RasterizedGlyph {
     // Calculate the cell metrics
     let height = metrics.line_height as i32 + i32::from(offset_y);
     let mut width = metrics.average_advance as i32 + i32::from(offset_x);
-    let line_width = cmp::max((f64::from(width) * CURSOR_WIDTH_PERCENTAGE).round() as i32, 1);
+    let cursor_thickness = if cursor == CursorStyle::HollowBlock {
+        CursorThickness::default().0
+    } else {
+        cursor_thickness.0
+    };
```

Comment:
The conversation was purely about `Beam` and `Underline` on the #3526, so I've decided to use default value for `HollowBlock` in the end, however I feel like we can safely apply thickness to it as well, since it makes things kind of consistent.